utils/check.go

```diff
@@ -0,0 +1,54 @@
+/*
+   Copyright 2020 Docker Compose CLI authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/docker/compose-cli/api/context/store"
+	"github.com/docker/compose-cli/cli/config"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+)
+
+var ErrUnsupportedFlag = errors.New("unsupported flag")
+
+func CheckUnsupported(errs *multierror.Error, toCheck, expectedValue interface{}, commandName, msg string) *multierror.Error {
+	ctype := store.DefaultContextType
+	currentContext := config.GetCurrentContext("", config.ConfDir(), nil)
+	s, _ := store.New(config.ConfDir())
+	cc, _ := s.Get(currentContext)
+	if cc != nil {
+		ctype = cc.Type()
+	}
+	// Fixes type for posterior comparison
+	switch toCheck.(type) {
+	case *time.Duration:
+		if expectedValue == nil {
+			var nilDurationPtr *time.Duration
+			expectedValue = nilDurationPtr
+		}
+	}
+	if toCheck != expectedValue {
```

Comment:
I think this would be more reliable
```suggestion
	if (toCheck != nil || expectedValue != nil) && toCheck != expectedValue {
```

Reply:
The problem here is that in GO, a literal `nil` is different of a typed `nil` when coming from a function parameter typed `interface{}`. 
That's why we need to check the type. If the expected is `nil` (literal), we need to check the type of the `toCheck` and change the expected to the same type. Even if the value is also `nil`.

See https://play.golang.org/p/vG12aUX1kvk

Reply:
Yes, which is what the `(toCheck != nil || expectedValue != nil)` was supposed to cover, but it’s a bit trickier than that.
I’m merely concerned that this isn’t very generic and will only correctly handle `time.Duration`.
Here is a generic variant: https://play.golang.org/p/mOf1bwAFgN0

Reply:
Cool! This is what I did at the beginning! But I just considered that was too complicated to handle just one type.
But if you don't think so I agree to implement it like that! 😸

Reply:
I think it’s probably fine. We can move it to `utils` as well.